plugins/provisioners/chef/provisioner/chef_solo.rb

```diff
@@ -72,8 +79,10 @@ def expanded_folders(paths, appended_folder=nil)
               local_path = File.expand_path(path, @machine.env.root_path)
 
               if File.exist?(local_path)
-                # Path exists on the host, setup the remote path
-                remote_path = "#{@config.provisioning_path}/chef-solo-#{get_and_update_counter(:cookbooks_path)}"
+                # Path exists on the host, setup the remote path. We use
+                # the MD5 of the local path so that it is predictable.
+                key         = Digest::MD5.hexdigest(local_path)
+                remote_path = "#{@config.provisioning_path}/#{key}"
```

Comment:
Probably, but I think its better for the feature to work as advertised. I'll make a note of it in the CHANGELOG.